### DISCUSSION TITLE
close n-dropdown programmitically?

### DISCUSSION THREAD

**Discussion by summerset-zz** [tusen-ai/naive-ui#3549] [Q&A] [answered]
I'm using n-dropdown with render feature, the code is like this:
```javascript
// when not logged in, render an n-button
         options.push({
            key: 'login',
            type: 'render',
            render: () => h(NButton, {
                class: 'titlebar-dropdown-link',
                onClick: () => {
                    util.jumpToPage('special:login?returnto=social', '_blank', 'club')
                },
                text: true
            },
                ['登录'])
        })
// when logged in, render a router-link
        options.push({
            key: 'mypage',
            type: 'render',
            render: () => h(RouterLink,
                {
                    class: 'titlebar-dropdown-link',
                    to: {
                        name: 'user',
                        params: {
                            userid: store.currentUser.profile?.id
                        }
                    }
                },
                ['我的空间']
            )
        })
```
the route jumps fine, but I have to click outside to close the dropdown. is there anyway to close the dropdown programmatically?

**Comment by 07akioni** [comment, marked as answer]
We should expose `setShow` method (I'll add it in next version)

**Reply by bkis** [reply]
Will this still happen at some point? I am having the same problem. Thanks!

**Comment by bkis** [comment]
Also, it seems like `on-select` is not emitted if a `DropdownRenderOption` is selected. Is it intentional that those aren't selectable? If one could select `DropdownRenderOption`s, it would also solve the problem, I guess.

**Comment by bkis** [comment]
I am now using `dropdownRef.value.doUpdateShow(false)` in my click handler, but I think this isn't part of the public API :(
